### ISSUE TITLE
DNS Rewrite TLD preserving subdomain (e.g. *.lan to *.fritz.box)

### ISSUE THREAD

**Issue by rdsubhas** [AdguardTeam/AdGuardHome#4538]
### Prerequisites

- [x] Discussed and suggested to file as feature request: https://github.com/AdguardTeam/AdGuardHome/discussions/4532
- [x] I am running the latest version
- [x] I checked the documentation and found no answer
- [x] I checked to make sure that this issue has not already been filed

### Problem Description
Many of us have routers which have non-configurable LAN domain names. e.g. `fritz.box`

We still use the router's DHCP, because it's helpful with router features (like setting up separate client zones, firewall rules, etc). Let's say we don't want to switch to AGH DHCP just for the lan domain name.

Can we rewrite `*.lan` to `*.fritz.box` using AGH, i.e. rewrite the TLD preserving subdomain? It would help all LAN devices internally talk via `<device-name>.lan` domain, but the address resolution can still be done by the upstream router as `<device>.fritz.box`.

### Proposed Solution
Option to "Preserve Subdomain" in the AdguardHome "DNS Rewrites" screen.

### Alternatives Considered
Not aware of any other alternatives.

**Comment by agneevX** [comment]
Yes, `local_domain_name`. [RTFM](https://github.com/AdguardTeam/AdGuardHome/wiki/Configuration#configuration-file).

**Comment by rdsubhas** [comment]
@agneevX read the issue description.

**Comment by agneevX** [comment]
My apologies. It's just that we get a ton of issues like this from users that is very similar to the plethora of features already available. 

While I think your request is a bit of an edge case, I nonetheless wish you good luck on getting it implemented.

**Comment by rdsubhas** [comment]
Thanks @agneevX for being transparent, my apologies too 👍🙂

**Comment by kelke** [comment]
I would love this feature as well. Is it planned in any shape or form?
